File: 44주차/모음사전.py
dict = {}
nth = 1
def solution(word):
   
    def dfs(res):
        global dict,nth
        if len(res)>=1:
            dict[res] = nth
            nth+=1
        if len(res) == 5:
            return
        for v in ['A','E','I','O','U']:
            dfs(res+v)
    dfs('')
    return dict[word]
# 프로그래머스 완전탐색.
# 모음 사전 문제.

File: 68주차/모음사전.js
function solution(word) {
  var answer = 0;
  const dict = {};
  let cnt = 1;
  const arr = ["A", "E", "I", "O", "U"];

  const dfs = (res) => {
    if (res.length > 5) return;
    if (res.length > 0) {
      dict[res] = cnt;
      cnt += 1;
    }
    for (const v of arr) {
      dfs(res + v);
    }
  };
  dfs("");

  return dict[word];
}
